custom/icds_reports/migrations/sql_templates/database_views/disha_indicators.sql

```diff
@@ -0,0 +1,92 @@
+        ELSE 100 * "ccs_record"."anc4_received_at_delivery"/"ccs_record"."delivered_in_month" END as "anc4_received_at_delivery_percent",
+    CASE WHEN "ccs_record"."pregnant"=0 THEN null
+        ELSE 100 * "ccs_record"."resting_during_pregnancy"/"ccs_record"."pregnant" END as "resting_during_pregnancy_percent",
+    CASE WHEN "ccs_record"."pregnant"=0 THEN null
+        ELSE 100 * "ccs_record"."extra_meal"/"ccs_record"."pregnant" END as "extra_meal_percent",
+    CASE WHEN "ccs_record"."trimester_3"=0 THEN null
+        ELSE 100 * "ccs_record"."counsel_immediate_bf"/"ccs_record"."trimester_3" END as "counsel_immediate_bf_percent",
+    CASE WHEN "child_health"."wer_eligible"=0 THEN null 
+        ELSE 100 * "child_health"."nutrition_status_weighed"/"child_health"."wer_eligible" END as "nutrition_status_weighed_percent",
+    CASE WHEN "child_health"."height_eligible"=0 THEN null 
+        ELSE 100 * "child_health"."height_measured_in_month"/"child_health"."height_eligible" END as "height_measured_in_month_percent",
+    nutrition_status_unweighed,
+    CASE WHEN "child_health"."nutrition_status_weighed"=0 THEN null 
+        ELSE 100 * "child_health"."nutrition_status_severely_underweight"/"child_health"."nutrition_status_weighed" END as "nutrition_status_severely_underweight_percent",
+    CASE WHEN "child_health"."nutrition_status_weighed"=0 THEN null 
+        ELSE 100 * "child_health"."nutrition_status_moderately_underweight"/"child_health"."nutrition_status_weighed" END as "nutrition_status_moderately_underweight_percent",
+    CASE WHEN "child_health"."weighed_and_height_measured_in_month"=0 THEN null 
+        ELSE 100 * "child_health"."wasting_severe"/"child_health"."weighed_and_height_measured_in_month" END as "wasting_severe_percent",
+    CASE WHEN "child_health"."weighed_and_height_measured_in_month"=0 THEN null 
+        ELSE 100 * "child_health"."wasting_moderate"/"child_health"."weighed_and_height_measured_in_month" END as "wasting_moderate_percent",
+    CASE WHEN "child_health"."height_measured_in_month"=0 THEN null 
+        ELSE 100 * "child_health"."stunting_severe"/"child_health"."height_measured_in_month" END as "stunting_severe_percent",
+    CASE WHEN  "child_health"."height_measured_in_month"=0 THEN null 
+        ELSE 100 * "child_health"."stunting_moderate"/"child_health"."height_measured_in_month" END as "stunting_moderate_percent",
+    CASE WHEN "child_health"."weighed_and_born_in_month"=0 THEN null 
+        ELSE 100 * "child_health"."low_birth_weight_in_month"/"child_health"."weighed_and_born_in_month" END as "low_birth_weight_in_month_percent",
+    CASE WHEN  "child_health"."fully_immunized_eligible"=0 THEN null 
+        ELSE 100 * ("child_health"."fully_immunized_on_time" + "child_health"."fully_immunized_late")/ "child_health"."fully_immunized_eligible" END as "immunized_percent"    
+FROM "agg_awc_monthly" "awc_monthly"
+LEFT JOIN "agg_ccs_record_monthly" "ccs_record" ON (
```

Comment:
Im pretty late to this review, but why are you joining views instead of the underlying tables? Many of the views include the same underlying tables, will this join those tables multiple times?

Reply:
(Would have been nice to get this input at an earlier point, for e.g. when we discussed this join approach or during the initial commits)

As far as I know nested database views allow reuse and isn't an anti-pattern (some of the icds views are nested). I am not totally sure whether they would recompute base views, but I suspect not @emord do you know? I will revisit this tomorrow if this is a concern. Otherwise, if it's g2g can you merge please.

Reply:
I think joining views is probably fine as long as the number of joins are minimized. Would be good if you could provide an EXPLAIN output to verify that it is correctly using indexes/not doing anything bad

Reply:
I think the query plan is using indices. Here's the [full plan](https://gist.github.com/sravfeyn/62511b8a851b3a75a4b03f1874f56da1). This is the minimum amount of joins required for this, since all these indicators are required in one dump file unfortunately. If the performance seems to be bad, we can latter iterate with the external team on possibly splitting data into separate dumps by table.

Reply:
This is a ginormous plan. I don't think it would complete at all. It does a sequential scan on all of the agg_awc tables. example:

```
)
                           ->  Seq Scan on "agg_awc_2017-12-01_1" agg_awc_43  (cost=0.00..2.11 rows=11 width=228)
                           ->  Seq Scan on "agg_awc_2017-12-01_2" agg_awc_44  (cost=0.00..11.59 rows=59 width=146)
                           ->  Seq Scan on "agg_awc_2017-12-01_3" agg_awc_45  (cost=0.00..84.89 rows=489 width=175)
                           ->  Seq Scan on "agg_awc_2017-12-01_4" agg_awc_46  (cost=0.00..658.78 rows=4878 width=204)
                           ->  Seq Scan on "agg_awc_2017-12-01_5" agg_awc_47  (cost=0.00..71315.40 rows=148040 width=233)
                           ->  Seq Scan on "agg_awc_2018-01-01_1" agg_awc_48  (cost=0.00..2.11 rows=11 width=228)
                           ->  Seq Scan on "agg_awc_2018-01-01_2" agg_awc_49  (cost=0.00..11.59 rows=59 width=146)
                           ->  Seq Scan on "agg_awc_2018-01-01_3" agg_awc_50  (cost=0.00..88.90 rows=490 width=175)
                           ->  Seq Scan on "agg_awc_2018-01-01_4" agg_awc_51  (cost=0.00..690.39 rows=4639 width=204)
                           ->  Seq Scan on "agg_awc_2018-01-01_5" agg_awc_52  (cost=0.00..80288.32 rows=154132 width=233)
                           ->  Seq Scan on "agg_awc_2018-02-01_1" agg_awc_53  (cost=0.00..2.11 rows=11 width=228)
                           ->  Seq Scan on "agg_awc_2018-02-01_2" agg_awc_54  (cost=0.00..11.59 rows=59 width=146)
                           ->  Seq Scan on "agg_awc_2018-02-01_3" agg_awc_55  (cost=0.00..90.90 rows=490 width=175)
                           ->  Seq Scan on "agg_awc_2018-02-01_4" agg_awc_56  (cost=0.00..726.85 rows=5285 width=204)
                           ->  Seq Scan on "agg_awc_2018-02-01_5" agg_awc_57  (cost=0.00..83502.00 rows=167900 width=233)
```

You will probably want to at least split up the back end aggregation on our side even if we have to send them one file  per state in the end

Reply:
offlined about this with @sravfeyn the original EXPLAIN did not include filters which is why it's so large. when they're provided it's correctly filtering table and using indexes

Reply:
(Offlined)

The above explain I had generated was for full view without month/state filtering. Here is the one with [filters](https://gist.github.com/sravfeyn/6b428e83bc4d34c057d693e70eaee322) and it looks good (filters to only month tables and uses index scans) .